snapshot-labs/sx-monorepo
# feat: set proposal app from query param

### Summary



Toward https://github.com/snapshot-labs/workflow/issues/235
Discussion https://discord.com/channels/955773041898573854/1296435871431786587

The current code is passing a hardcoded `snapshot-v2` as `app`, when creating an offchain proposal.

This PR will: 

- use the `app` query param from the Editor when available, and use `snapshot-v2` as fallback.
- reflect the same changes for ethereum and starknet `propose` action

### How to test

1. Go to a proposal editor (try on offchain and onchain)
2. append the `app` query param to the url: `?app=test-pr`
3. Create a proposal
4. In the signature payload, you should see the `app` property set to `test-pr`
5. View your proposal metadata
6. `app` should be set to `test-pr`
7. When missing, `app` should be set to `snapshot-v2`

### Note

- when `app` is too long (>128chars), it will be ignored, and default value will be used
